WooCommerce/Classes/ViewRelated/Products/Edit Product/Edit Price/ProductPriceSettingsViewController.swift

```diff
@@ -333,17 +333,57 @@ private extension ProductPriceSettingsViewController {
     func configurePrice(cell: UnitInputTableViewCell) {
         let cellViewModel = Product.createRegularPriceViewModel(regularPrice: viewModel.regularPrice, using: CurrencySettings.shared) { [weak self] value in
             self?.viewModel.handleRegularPriceChange(value)
+            self?.setCellPriceAccessibility(price: cell.value, cell: cell)
         }
         cell.selectionStyle = .none
         cell.configure(viewModel: cellViewModel)
+        setCellPriceAccessibility(price: cell.value, cell: cell)
+    }
+
+    func setCellPriceAccessibility(price: String?, cell: UnitInputTableViewCell) {
+        let value = cell.value.isEmpty == false ? cell.value : NSLocalizedString("empty", comment: "Accessibility text for when a price is empty")
+        let currency = cell.value.isEmpty == false ? CurrencySettings.shared.currencyCode.rawValue : ""
+        let accessibilityText = String.localizedStringWithFormat(
+            NSLocalizedString(
+                "Price %1$@ %2$@",
+                comment: "Accessibility label for the regular price cell in Product Price. It's read like: Price 100 dollars."
+            ),
+            value,
+            currency
+        )
+        let accessibilityHint = NSLocalizedString(
+            "The regular price for this product, which is also editable.",
+            comment: "VoiceOver accessibility hint, informing the user that the cell shows the price information for this product."
+        )
+        cell.setAccessitibily(label: accessibilityText, hint: accessibilityHint)
     }
 
     func configureSalePrice(cell: UnitInputTableViewCell) {
         let cellViewModel = Product.createSalePriceViewModel(salePrice: viewModel.salePrice, using: CurrencySettings.shared) { [weak self] value in
             self?.viewModel.handleSalePriceChange(value)
+            self?.setCellSalePriceAccessibility(price: cell.value, cell: cell)
         }
         cell.selectionStyle = .none
         cell.configure(viewModel: cellViewModel)
+        setCellSalePriceAccessibility(price: cell.value, cell: cell)
+    }
+
+    func setCellSalePriceAccessibility(price: String?, cell: UnitInputTableViewCell) {
```

Comment:
What do you think about moving most of these to `UnitInputTableViewCell` instead? 

Typically, accessibility property settings are _reactive_. They are mostly only _executed_ when an iOS accessibility setting is turned on. I think it would make sense if we can follow that trend. [This MonthNavigationView example](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/docs/voiceover-guidelines.md#consider-using-adjustables-to-consolidate-related-buttons) shows how overriding accessibility properties can clarify how the accessibility configuration of a _custom element_ is set up. 

I think this would also make `UnitInputTableViewCell` work well by default without any configuration. 

```swift

final class UnitInputTableViewCell: UITableViewCell {


    /// Make this the accessibile element. Do not allow text field to be accessible right away.
    override var isAccessibilityElement: Bool {
        get {
            true
        }
        set { }
    }

    override var accessibilityLabel: String? {
        get {
            titleLabel.text
        }
        set { }
    }

    override var accessibilityValue: String? {
        get {
            if inputTextField.text?.isEmpty ?? true {
                return "Empty"
            } else {
                // This should probably be localized
                return (inputTextField.text ?? "") + " " + (unitLabel.text ?? "")
            }
        }
        set { }
    }

    /// Make text field the first responder when the user double-taps on the cell.
    override func accessibilityActivate() -> Bool {
        inputTextField.becomeFirstResponder()
        return true
    }

}
```

I think this eliminates some code since we're reusing the `titleLabel` and `inputTextField` values. Making these reactive would also make sure that we _don't forget_ to update them when the `configure()` method is called. There are some things that I have not verified how they're going to work like localization. 😅 

As for the accessibility hint, maybe that can be passed as part of the `UnitInputViewModel`? We can _contain_ the “Editable“ part inside `UnitInputTableViewCell` but let the ViewModel customize the first sentence. Something like this: 

```swift
    override var accessibilityHint: String? {
        get {
            let format = NSLocalizedString("%@. Editable", comment: "...")
            return String.localizedStringWithFormat(format, viewModel.accessibilityHint)
        }
        set { }
    }
```

Disclaimer: I have not tested this fully with other usages of `UnitInputTableViewCell`. 😄

Reply:
Thanks @shiki for all this precious information! I really like your proposal.

```
override var isAccessibilityElement: Bool {
        get {
            true
        }
        set { }
    }
```
This is a trick I didn't know for disabling all the elements nested inside it, and it makes things easier.

```
// This should probably be localized
                return (inputTextField.text ?? "") + " " + (unitLabel.text ?? "")
```

I don't think this needs localization, but we can't use `unitLabel` for the price since it will show up the currency symbol `$`, and not the currency code (like `usd` witch will be read as dollars). Do you think that makes sense to add as part of the `UnitInputViewModel`, a specific variable for configuring the accessibility value of `unitLabel`?



> I have not tested this fully with other usages of UnitInputTableViewCell

If well implemented, I think this covers all the cases. I can make a list of all the use case that needs to be tested. 👍

Reply:
> Do you think that makes sense to add as part of the UnitInputViewModel, a specific variable for configuring the accessibility value of unitLabel?

@pmusolino That's the one I wasn't sure about. I saw you used the `USD` or `CAD` variants but using the `unitLabel.text` worked for me when I tried it. VoiceOver spoke the value and currency naturally. 

I'd trust your judgement more here though because I have less experience with `UnitInputTableViewCell`. 🙂  If you think using the `unitLabel` is not a good idea then I'm totally fine with adding a `unitAccessibilityName` (or something) in the `ViewModel`. 

Also, note that my proposed code was just a quick mock to relay the code structure. So not a lot of thought was spent on testing other scenarios. 😅

Reply:
> VoiceOver spoke the value and currency naturally.

Oh, I see. Probably when the VoiceOver reads the symbol directly from the label, it pronounces the symbol and not the value. Adding it in a context seems to solve the issue!